open-grant-proposals/ars-go-filecoin-api.md

```diff
@@ -0,0 +1,110 @@
+# Open Grant Proposal: `go-filecoin-api-client`
+
+**Name of Project:** `go-filecoin-api-client`
+
+**Proposal Category:** `devtools-libraries`
+
+**Proposer:** 	[@waynewyang](https://github.com/waynewyang)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?** 	"Yes"
+
+# Project Description
+
+Currently, filecoin have not provided the corresponding filecoin-api of go version. Every developer must implement their own set of code logic to access filecoin when using go-lang.
+
+In order to solve the above problems, avoid developers from manufacturing wheels repeatedly and improve development efficiency, a complete set of go version filecoin-api can be developed. Through various packages of SDK, third-party or specific function implementation can be effectively blocked, so as to reduce the cost of developers' understanding of functions and improve access efficiency.
+
+## Value
+
+Currently, filecoin ecology does not have a go version of the API, which can greatly improve developers' access efficiency.
+
+- What are the risks if you don't get it right?
+
+​	Once released, anything will affect anyone who USES the release package, so doing it wrong will affect anyone who USES the SDK.
+
+- What are the risks that will make executing on this project difficult?
+
+​	How to design a general extensible SDK architecture, easy to understand the naming specification.
+
+## Deliverables
+
+## Development Roadmap
+
+| Milestone No. | Milestone Description                                        | Funding | Estimated Timeframe                        |
+| ------------- | ------------------------------------------------------------ | ------- | ------------------------------------------ |
+| 1             | design and Implement                                         | $2000   | Three weeks                                |
+| 2             | All project deliverables are completed and added to the [Filecoin Shipyard](https://github.com/filecoin-shipyard) and Maintenance | $500    | One week<br />+<br />Long time maintenance |
+
+
+
+For each milestone, please describe:
+
+**M1:**
+
+
+| No   | Description          | Team Member                         | Funding Required |
+| ---- | -------------------- | ----------------------------------- | ---------------- |
+| 1    | design and Implement | @johnli-helloworld<br />@waynewyang | $2000            |
+
+**M2:**
+
+| No   | Description                                                  | Team Member | Funding Required |
+| ---- | ------------------------------------------------------------ | ----------- | ---------------- |
+| 1    | All project deliverables are completed and added to the [Filecoin Shipyard](https://github.com/filecoin-shipyard) and Maintenance | Arsyun Team | $500             |
```

Comment:
Does this project include documentation?

Reply:
yes